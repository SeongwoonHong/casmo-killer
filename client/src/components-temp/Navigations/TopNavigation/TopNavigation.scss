.top-nav-btn {
    @include rectanglify($top-nav-size);
    background-color: transparent;
    border: none;
    color: #fff;
    padding: 0;
    &:focus {
        background-color: transparent;
    }
    > .material-icons {
        font-size: 24px;
        height: 100%;
        line-height: $top-nav-size;
    }
}
a.top-nav-btn {
    text-align: center;
}
.navigation-headers {
    background-color: palette(Teal, 800);
    border-bottom: 1px solid $teal-1000;
    height: $top-nav-size;
}
h1 {
    font-size: 20px;
    font-weight: 700;
    margin: 0;
    text-align: center;
    width: $main-menu-width - $top-nav-size;
    > a {
        color: #fff;
        display: block;
        line-height: $top-nav-size;
    }
}
.Top-navigation {
    @include flex-box(space-between);
    .notification {
        position: relative;
        .notification-badge {
            background-color: palette(Red, 500);
            color: #fff;
            border-radius: 50%;
            font-size: 10px;
            line-height: normal;
            padding: 1px 3px;
            position: absolute;
        }
    }
}
