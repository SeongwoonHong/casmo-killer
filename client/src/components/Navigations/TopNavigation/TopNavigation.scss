.Top-navigation {
    background-color: palette(Grey, 800);
    border-bottom: 1px solid palette(Grey, 900);
    height: $top-nav-size;
    &__row {
        @include flex-box(space-between);
    }
    @media screen and (min-width: 992px) {
        background:
            linear-gradient(
                rgba(0, 0, 0, 0.7),
                rgba(0, 0, 0, 0.7)
            ),
            url('../../../assets/header-bg.png') no-repeat center;
        background-size: cover;
        height: 75px;
        &__row {
            padding: 0 15px;
        }
        &__btn {
            display: none;
        }
    }
    @media screen and (min-width: 1220px) {
        &__row {
            padding: 0;
        }
    }
}
