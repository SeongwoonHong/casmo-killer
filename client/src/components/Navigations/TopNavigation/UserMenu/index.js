import { connect } from 'react-redux';
import * as actions from 'actions';

import UserMenu from './UserMenu';

const mapStateToProps = (state) => {
  return {
    layout: state.layout,
    user: state.user
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    toggleUserDropdown: isOpen => dispatch(actions.toggleUserDropdown(isOpen)),
    openAuthModal: authType => dispatch(actions.openAuthModal(authType))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(UserMenu);
