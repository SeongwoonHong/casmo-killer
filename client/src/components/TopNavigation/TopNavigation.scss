.top-main-nav {
    align-items: center;
    background-color: #004d40;
    display: flex;
    height: 50px;
    justify-content: space-between;
    .btn {
        background-color: transparent;
        border-radius: 0;
        box-shadow: none;
        height: 50px;
        padding: 0 13px;
        position: relative;
        transition: all .3s;
        vertical-align: baseline;
        &.active {
            background-color: #004d40;
            position: absolute;
            transform: translateX(270px);
            z-index: 99;
        }
        > .material-icons {
            color: #e0f2f1;
            font-size: 24px;
            height: 100%;
            line-height: 50px;
        }
        > .notification-badge {
            color: #e0f2f1;
            background-color: #ff3d00;
            border: 1px solid #e0f2f1;
            border-radius: 50%;
            font-size: 11px;
            line-height: normal;
            padding: 1px 4px;
            position: absolute;
            right: 0;
            top: 10px;
        }
    }
    > h1 {
        font-size: 20px;
        font-weight: 700;
        margin: 0;
        text-align: center;
        > a {
            color: #e0f2f1;
            display: block;
            line-height: 50px;
        }
    }
    .btn.hide-in-mobile,
    > .search-form-dt {
        display: none;
    }
    @media screen and (min-width: 360px) {
        .btn {
            &.active {
                transform: translateX(310px);
            }
        }
    }
    @media screen and (min-width: 768px) {
        height: 60px;
        .btn {
            height: 60px;
            padding: 0 18px;
            &.hide-in-mobile {
                display: inline-block;
            }
            > .material-icons {
                line-height: 60px;
            }
        }
        > h1 {
            font-size: 24px;
            > a {
                line-height: 60px;
            }
        }
    }
    @media screen and (min-width: 992px) {
        .btn {
            &.hide-in-dt {
                display: none;
            }
        }
        > h1 {
            width: 260px;
        }
        > .search-form-dt {
            background-color: #00695c;
            display: flex;
            height: 40px;
            margin-left: auto;
            margin-right: 10px;
            padding: 0 15px;
            width: 300px;
            > .material-icons {
                color: #e0f2f1;
                line-height: 40px;
                margin-right: 15px;
            }
            > input[type=text]:not(.browser-default) {
                border: none;
                color: #e0f2f1;
                height: auto;
                margin: 0;
                &:focus:not([readonly]) {
                    box-shadow: none;
                }
                &::placeholder {
                    color: #e0f2f1;
                }
            }
        }
    }
}
