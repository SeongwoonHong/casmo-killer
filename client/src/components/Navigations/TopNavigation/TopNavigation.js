import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import './TopNavigation.scss';

import SearchForm from '../SearchForm/SearchForm';

class TopNavigation extends Component {

  render() {

    const { toggleMenu, toggleUserMenu } = this.props;

    return (
      <header className="Top-navigation nav-headers">
        <div className="component-row">
          <button
            type="button"
            className="top-nav-btn"
            onClick={ toggleMenu }>
            <i className="material-icons">
              dehaze
            </i>
          </button>
          <h1>
            <Link to="/community">
              CK BOARD
            </Link>
          </h1>
          <SearchForm styleClass="dt" />
          <button
            type="button"
            className="top-nav-btn"
            onClick={ toggleUserMenu }>
            <i className="material-icons">
              person_outline
            </i>
          </button>
        </div>
      </header>
    );

  }

}

export default TopNavigation;
