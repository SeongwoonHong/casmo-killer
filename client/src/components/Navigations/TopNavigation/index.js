import { connect } from 'react-redux';
import * as actions from 'actions';

import TopNavigation from './TopNavigation';

const mapStateToProps = (state) => {
  return {
    user: state.user,
    layout: state.layout
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    toggleMenu: () => dispatch(actions.toggleMenu()),
    toggleSearchForm: () => dispatch(actions.toggleSearchForm()),
    openAuthModal: authType => dispatch(actions.openAuthModal(authType)),
    toggleUserDropdown: isOpen => dispatch(actions.toggleUserDropdown(isOpen))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(TopNavigation);
