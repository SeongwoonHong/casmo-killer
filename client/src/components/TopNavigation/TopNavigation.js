import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import classnames from 'classnames';

import './TopNavigation.scss';

import SearchForm from '../SearchForm/SearchForm';
import UserMenu from './UserMenu';

class TopNavigation extends Component {

  render() {

    const { layout } = this.props;

    return (
      <header className="teal darken-4 z-depth-2 top-main-nav">
        <a
          role="button"
          tabIndex={ 0 }
          className={ classnames('btn teal darken-4 hide-in-dt', {
            toggled: layout.isMainMenuVisible
          }) }
          onClick={ this.props.toggleMenu }
          onKeyDown={ () => {} }>
          <i className="material-icons teal-text text-lighten-5">
            {
              layout.isMainMenuVisible
                ? 'close'
                : 'dehaze'
            }
          </i>
        </a>
        <h1>
          <Link
            to="/"
            className="teal-text text-lighten-5">
            CK BOARD
          </Link>
        </h1>
        <SearchForm styleClass="dt" />
        <UserMenu />
      </header>
    );

  }

}

export default TopNavigation;
