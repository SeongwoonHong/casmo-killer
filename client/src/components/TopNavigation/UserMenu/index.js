import { connect } from 'react-redux';
import UserMenu from './UserMenu';
import * as actions from '../../../actions';

const mapStateToProps = (state) => {
  return {
    layout: state.layout
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    toggleSearchForm: () => dispatch(actions.toggleSearchForm()),
    toggleMenu: () => dispatch(actions.toggleMenu())
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(UserMenu);
