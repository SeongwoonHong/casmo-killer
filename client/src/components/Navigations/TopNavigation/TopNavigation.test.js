import React from 'react';
import { shallow } from 'enzyme';

import TopNavigation from './TopNavigation';

describe('<TopNavigation />', () => {

  let component = null;

  beforeEach(() => {
    component = shallow(<TopNavigation />);
  });

  it('renders the component consistently', () => {
    expect(component).toMatchSnapshot();
  });

});
