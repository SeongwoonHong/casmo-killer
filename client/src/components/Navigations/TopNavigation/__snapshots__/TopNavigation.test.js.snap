// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopNavigation /> renders the component consistently 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopNavigation />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="Top-navigation__row component-row"
>
        <TopNavBtn
                className=""
                classPrefix="Top-navigation"
                icon="dehaze"
                onClick={[Function]}
        />
        <AppTitle />
        <TopNavBtn
                className=""
                classPrefix="Top-navigation"
                icon="person_outline"
                onClick={[Function]}
        />
</div>,
      "className": "Top-navigation",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TopNavBtn
            className=""
            classPrefix="Top-navigation"
            icon="dehaze"
            onClick={[Function]}
/>,
          <AppTitle />,
          <TopNavBtn
            className=""
            classPrefix="Top-navigation"
            icon="person_outline"
            onClick={[Function]}
/>,
        ],
        "className": "Top-navigation__row component-row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "",
            "className": "",
            "classPrefix": "Top-navigation",
            "icon": "dehaze",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "",
            "className": "",
            "classPrefix": "Top-navigation",
            "icon": "person_outline",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="Top-navigation__row component-row"
>
          <TopNavBtn
                    className=""
                    classPrefix="Top-navigation"
                    icon="dehaze"
                    onClick={[Function]}
          />
          <AppTitle />
          <TopNavBtn
                    className=""
                    classPrefix="Top-navigation"
                    icon="person_outline"
                    onClick={[Function]}
          />
</div>,
        "className": "Top-navigation",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TopNavBtn
              className=""
              classPrefix="Top-navigation"
              icon="dehaze"
              onClick={[Function]}
/>,
            <AppTitle />,
            <TopNavBtn
              className=""
              classPrefix="Top-navigation"
              icon="person_outline"
              onClick={[Function]}
/>,
          ],
          "className": "Top-navigation__row component-row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "",
              "className": "",
              "classPrefix": "Top-navigation",
              "icon": "dehaze",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "",
              "className": "",
              "classPrefix": "Top-navigation",
              "icon": "person_outline",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
